src/ILCompiler.Compiler/src/IL/Stubs/StartupCode/StartupCodeMainMethod.cs

```diff
@@ -56,6 +59,16 @@ public override MethodIL EmitIL()
             ILEmitter emitter = new ILEmitter();
             ILCodeStream codeStream = emitter.NewCodeStream();
 
+            // Allow the class library to run explicitly ordered class constructors first thing in start-up.
+            if (_explicitlyRunCctors != null)
+            {
+                foreach (TypeDesc type in _explicitlyRunCctors)
+                {
+                    MethodDesc cctor = type.GetMethod(".cctor", null);
+                    codeStream.Emit(ILOpcode.call, emitter.NewToken(cctor));
+                }
+            }
+            
             ModuleDesc developerExperience = Context.ResolveAssembly(new AssemblyName("System.Private.DeveloperExperience.Console"), false);
```

Comment:
Can the initialization of the developer experience be on the same plan - be just one of the number of potential entries to initialize?